sign.go

```diff
@@ -112,6 +86,46 @@ func (file *File) MarshalForSignature() ([]byte, error) {
 	return file.Marshal()
 }
 
+// Hash takes an input and a signature algorithm and returns its hashed value
+func Hash(input []byte, sigalg uint32) (output []byte, h crypto.Hash, err error) {
```

Comment:
might want a type alias for the sig algs

Reply:
does that buy us anything? we'll need to type assert to uint32 for marshalling

Reply:
it's more explicit